src/features/auth/refresh-token.ts

```diff
@@ -1,25 +1,42 @@
 import { errors, sbvrUtils } from '@balena/pinejs';
-import { checkUserPassword, getUser } from '../../infra/auth/auth';
+import { checkUserPassword, getUser, findUser } from '../../infra/auth/auth';
 import { updateUserXHR } from '../../infra/auth/jwt';
 import { captureException } from '../../infra/error-handling';
 import type { RequestHandler } from 'express';
 
-const { BadRequestError } = errors;
+import type { SetupOptions } from '../..';
+
+const { BadRequestError, NotFoundError } = errors;
+
+let onTokenRefresh: SetupOptions['onTokenRefresh'];
+export const setOnTokenRefresh = (
```

Comment:
This is done as a separate function because following the same pattern as "onLogin" would require a breaking change on `refreshToken` function which is exported and used for `/whoami`